tests/unit/s2n_fingerprint_ja4_test.c

```diff
@@ -690,12 +690,12 @@ int main(int argc, char **argv)
                             S2N_TEST_CLIENT_HELLO_CIPHERS,
                             S2N_TEST_CLIENT_HELLO_AFTER_CIPHERS,
                             /* extensions size */
-                            0x00, 24,
+                            0x00, 20,
                             /* extension: alpn */
-                            0x00, TLS_EXTENSION_ALPN, 0x00, 20,
-                            0x00, 18,
-                            0, 0x00, 8, 'h', 't', 't', 'p', '/', '1', '.', '1',
-                            0, 0x00, 4, 'q', 'u', 'i', 'c');
```

Comment:
Probably not worth doing now, and it might be too complex or not be possible for some tests, but I wonder if this could have been caught if these tests modified an s2n_connection and then called s2n_client_hello_send rather than manually constructing the client hello? Then any assumptions about the structure of the client hello wouldn't be in the tests themselves.